internal/charmstore/addentity.go

```diff
@@ -513,11 +513,12 @@ func (s *Store) addEntity(entity *mongodoc.Entity) (err error) {
 		Write: perms,
 	}
 	baseEntity := &mongodoc.BaseEntity{
-		URL:         entity.BaseURL,
-		User:        entity.User,
-		Name:        entity.Name,
-		Public:      false,
-		ACLs:        acls,
+		URL:  entity.BaseURL,
+		User: entity.User,
+		Name: entity.Name,
+		ChannelACLs: map[mongodoc.Channel]mongodoc.ACL{
+			mongodoc.UnpublishedChannel: acls,
```

Comment:
I wonder if we should just set the initial acls for all channels here, so that when you publish for the first time we already have a reasonable default (user both read and write).